Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Task.java

```diff
@@ -396,14 +410,79 @@ public void setUsers(List<User> users) {
 
     /**
      * Set list of user groups.
-     * 
+     *
      * @param userGroups
      *            as list
      */
     public void setUserGroups(List<UserGroup> userGroups) {
         this.userGroups = userGroups;
     }
 
+    /**
+     * Get list of folders whose contents are to be generated.
+     *
+     * @return list of Folder objects or empty list
+     */
+    public List<Folder> getContentFolders() {
+        if (this.contentFolders == null) {
+            this.contentFolders = new ArrayList<>();
+        }
+        return contentFolders;
+    }
+
+    /**
+     * Set list of folders whose contents are to be generated.
+     *
+     * @param contentFolders
+     *            as list
+     */
+    public void setContentFolders(List<Folder> contentFolders) {
+        this.contentFolders = contentFolders;
+    }
+
+    /**
+     * Get list of switch objects for all folders whose contents can be
+     * generated.
+     *
+     * @return list of GeneratorSwitch objects or empty list
+     */
+    @SuppressWarnings("serial")
+    public List<GeneratorSwitch> getGenerators() {
+        if (this.contentFolders == null) {
```

Comment:
Logic inside bean class - it needs to be moved to other place - advisable TaskService.